core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -393,6 +394,15 @@ class RuntimeInfo private[spark](
     val javaHome: String,
     val scalaVersion: String)
 
+class FailureReason private[spark](
+    val failureType: String,
+    val message: String,
```

Comment:
The first line of `stackTrace` seems to the same with `message` so it's redundant.

Reply:
It's not always the same, see below comment for detailed explanation.